pkg/cloudprovider/providers/vsphere/vsphere.go

```diff
@@ -100,19 +103,49 @@ func vsphereLogin(cfg *VSphereConfig, ctx context.Context) (*govmomi.Client, err
 	return c, nil
 }
 
-type Instances struct {
-	cfg *VSphereConfig
+func getVirtualMachineByName(cfg *VSphereConfig, ctx context.Context, c *govmomi.Client, name string) (*object.VirtualMachine, error){
+	// Create a new finder
+	f := find.NewFinder(c.Client, true)
+
+	// Fetch and set data center
+	dc, err := f.DatacenterOrDefault(ctx, cfg.Global.Datacenter)
```

Comment:
this should be Datacenter(ctx, cfg.Global.Datacenter) , DatacenterOrDefault() returns default dc if its unable to find the given datacenter.